### ISSUE TITLE
Couldn't fetch information : Cannot read properties of undefined (reading 'split')

### ISSUE THREAD

**Issue by arpitgoyall** [dvishal485/flipkart-scraper-api#11]
The issue I found when we try to scrape the link of any mobile product:

```
{
"error": "Couldn't fetch information : Cannot read properties of undefined (reading 'split')",
"possible_solution": "Don't lose hope, contact the support",
"bug_report": "https://github.com/dvishal485/flipkart-scraper-api/issues"
}
```

This is the link I tested :  https://www.flipkart.com/realme-11x-5g-midnight-black-128-gb/p/itm07be1a2ff1a1b?pid=MOBGS2W3KYRQF4GZ

**Comment by devnydv** [comment]
https://flipkart.dvishal485.workers.dev/product/boult-audio-w40-quad-mic-enc-48h-battery-life-low-latency-gaming-made-india-5-3v-bluetooth-headset/p/itm3988812dde8cb
facing the same issue, for some products pid is also required in url param

**Comment by dvishal485** [comment]
I'm busy for the next few weeks, and this probably requires addition of additional branching to scraping logic. I'll definitely review the issue later on.
For now, you can try to patch it by yourself if possible (debug to find the source of error) or you'll have to wait for sometime.

**Comment by dvishal485** [comment]
> The issue I found when we try to scrape the link of any mobile product:
> 
> ```
> {
> "error": "Couldn't fetch information : Cannot read properties of undefined (reading 'split')",
> "possible_solution": "Don't lose hope, contact the support",
> "bug_report": "https://github.com/dvishal485/flipkart-scraper-api/issues"
> }
> ```
> 
> This is the link I tested : https://www.flipkart.com/realme-11x-5g-midnight-black-128-gb/p/itm07be1a2ff1a1b?pid=MOBGS2W3KYRQF4GZ

The issue was URL parameters (pid in this case) weren't being passed to the function which fetches the URL. This has been fixed with the commit cbdf05efb829556ff3d41453b844bd6a1ceefed2.

Here is the corresponding API URL to test: https://flipkart-scraper-api.dvishal485.workers.dev/product/realme-11x-5g-midnight-black-128-gb/p/itm07be1a2ff1a1b?pid=MOBGS2W3KYRQF4GZ

**Comment by dvishal485** [comment]
> https://flipkart.dvishal485.workers.dev/product/boult-audio-w40-quad-mic-enc-48h-battery-life-low-latency-gaming-made-india-5-3v-bluetooth-headset/p/itm3988812dde8cb facing the same issue, for some products pid is also required in url param

The URL mentioned here corresponds to a wrong product. After searching the product on Flipkart, I was able to find the correct product link which is scrapped properly as expected.

Here is the link to the API: https://flipkart-scraper-api.dvishal485.workers.dev/product/boult-audio-w40-quad-mic-enc-48h-battery-life-low-latency-gaming-made-india-5-3v-bluetooth-headset/p/itm5bd43329ddab7?pid=ACCGRNFYNKWSPC6D

Let me know if you face any other issue here, so I can reopen the issue. As per my knowledge this issue should be fixed now.
